pkg/ipmi/set_session_privilege_level.go

```diff
@@ -0,0 +1,84 @@
+package ipmi
+
+import (
+	"fmt"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+// SetSessionPrivilegeLevelReq implements the Set Session Privilege Level command, specified in section
+// 18.16 of v1.5 and 22.18 of v2.0.
+type SetSessionPrivilegeLevelReq struct {
+	layers.BaseLayer
+
+	// PrivilegeLevel indicates the privilege level to switch to.
+	// Set to 0 to retrieve the present privilege level.
+	PrivilegeLevel PrivilegeLevel
+}
+
+func (*SetSessionPrivilegeLevelReq) LayerType() gopacket.LayerType {
+	return LayerTypeSetSessionPrivilegeLevelReq
+}
+
+func (c *SetSessionPrivilegeLevelReq) SerializeTo(b gopacket.SerializeBuffer, _ gopacket.SerializeOptions) error {
+	bytes, err := b.PrependBytes(1)
```

Comment:
Before prepending the bytes, let's add a validation check for `if c.PrivilegeLevel == PrivilegeLevelCallback`, which is reserved in the spec.